File: golang-server/internal/dto/responseFormat.go
package dto

import "net/http"

const (
	messageOK = "OK"
)

type ResponseFormat struct {
	Status  int         `json:"status"`
	Message string      `json:"message"`
	Content interface{} `json:"content,omitempty"`
}

func Success(content interface{}) ResponseFormat {
	return ResponseFormat{
		http.StatusOK,
		messageOK,
		content,
	}
}

func Fail(status int, message string) ResponseFormat {
	return ResponseFormat{
		status,
		message,
		nil,
	}
}

File: java-code/src/main/java/com/fiddich/model/ResponseFormat.java
package com.fiddich.model;

public class ResponseFormat<T> {
    private Integer status;
    private String message;
    private T content;

    public ResponseFormat() {}

    public Integer getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

    public T getContent() {
        return content;
    }
}

